lib/github_service/commands/cross_repo_test.rb

```diff
@@ -0,0 +1,278 @@
+module GithubService
+  module Commands
+    # = GithubService::Commands::CrossRepoTest
+    #
+    # Triggers a build given the configured test repo:
+    #
+    #   Settings.cross_repo_tests_repo.*
+    #
+    # Which doing so will:
+    #
+    #    - validate the command
+    #    - create a branch
+    #    - create a commit with the travis.yml changes
+    #    - push said branch to the origin
+    #    - create a pull request for that branch
+    #
+    # More info can be found here on how the whole process works:
+    #
+    #   https://github.com/ManageIQ/manageiq-cross_repo-tests
+    #
+    # == Command structure
+    #
+    # @miq-bot cross-repo-test [<repos-to-test>] [including <extra-repos>]
+    #
+    # where:
+    #
+    #   - a `repo` is of the form [org/]repo[@ref|#pr]
+    #   - `repos-to-test` is a list of repos to have tested
+    #   - `extra-repos` is a list of repos (gems) to override in the bundle
+    #
+    # each "lists of repos" should be comma delimited.
+    #
+    # == Example
+    #
+    # In ManageIQ/manageiq PR #1234
+    #
+    #   @miq-bot cross-repo-test manageiq-api,manageiq-ui-classic#5678 \
+    #     including Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature
+    #
+    # will create a commit with the .travis.yml changes:
+    #
+    #   @@ -14,6 +14,6 @@ matrix:
+    #      fast_finish: true
+    #    env:
+    #      global:
+    #   -  - REPOS=
+    #   +  - REPOS=Fryguy/more_core_extensions@feature,Fryguy/linux_admin@feature,ManageIQ/manageiq#1234,manageiq-ui-classic#5678
+    #      matrix:
+    #   -  - TEST_REPO=
+    #   +  - TEST_REPO=ManageIQ/manageiq-api
+    #   +  - TEST_REPO=ManageIQ/manageiq-ui-classic#5678
+    #
+    # TODO:  Handle the "self" case, where `manageiq` is also a TEST_REPO
+    #
+    # (maybe include a "self" helper as well?)
+    #
+    class CrossRepoTest < Base
+      # Reference to the branch we are creating off of origin/master
+      attr_reader :branch_ref
+
+      # The user calling the command
+      attr_reader :issuer
+
+      # The (extra) repo(s) being targeted to be included in the test run
+      attr_reader :repos
+
+      # The repo(s) that will have the test suite run
+      attr_reader :test_repos
+
+      # The *-cross_repo-tests rugged instance
+      attr_reader :rugged_repo
+
+      # The arguments for the `cross-repo-test` command being called
+      attr_reader :value
+
+      restrict_to :organization
+
+      def self.test_repo_url
+        Settings.cross_repo_tests.url
+      end
+
+      def self.test_repo_name
+        Settings.cross_repo_tests.name
```

Comment:
This is the only change that I made after @Fryguy just approved, which was just:

```diff
- Settings.run_tests_repo.name
+ Settings.cross_repo_tests.name
```